### DISCUSSION TITLE
Failed Request IRQ for GPIO

### DISCUSSION THREAD

**Discussion by AhmadPanogari** [linux-chenxing/linux-chenxing.org#79] [General]
I want to use rotary encode but there is an error like this when booting :

![image](https://user-images.githubusercontent.com/17040103/214991323-a69826b8-349c-4920-89f9-9c1aca41221a.png)

this is the Device Tree configuration that I use

	gpio-keys {
		compatible = "gpio-keys";
		pinctrl-names = "default";

		rotary0_button {
			label = "rotary0_button";
			gpios = <&gpio SSD20XD_GPIO_GPIO2 GPIO_ACTIVE_LOW>;
			debounce-interval = <10>;
			linux,code = <28>;
		};
	};

	rotary@0 {
		compatible = "rotary-encoder";
		gpios = <&gpio SSD20XD_GPIO_FUART_CTS GPIO_ACTIVE_HIGH>, <&gpio SSD20XD_GPIO_FUART_RTS GPIO_ACTIVE_LOW>; /* GPIO47 is inverted */
		linux,axis = <0>; /* REL_X */
		rotary-encoder,encoding = "gray";
		rotary-encoder,relative-axis;
	};
	

Hopefully it can be resolved.

**Comment by fifteenhex** [comment]
Does this need both edge irqs? I don't think the hardware can do them
